Mathlib/NumberTheory/LSeries/HurwitzZetaEven.lean

```diff
@@ -395,3 +407,406 @@ lemma completedCosZeta₀_one_sub (a : UnitAddCircle) (s : ℂ) :
+`s = 0` if `a ≠ 0`)
+-/
+lemma differentiableAt_completedHurwitzZetaEven
+    (a : UnitAddCircle) {s : ℂ} (hs : s ≠ 0 ∨ a ≠ 0) (hs' : s ≠ 1) :
+    DifferentiableAt ℂ (completedHurwitzZetaEven a) s := by
+  refine (((hurwitzEvenFEPair a).differentiableAt_Λ ?_ (Or.inl ?_)).comp s
+      (differentiableAt_id.div_const _)).div_const _
+  · rw [div_ne_zero_iff, eq_true_intro two_ne_zero, and_true]
+    refine Or.imp (by tauto) (fun ha ↦ ?_) hs
+    simp only [hurwitzEvenFEPair, eq_false_intro ha, if_false]
+  · change s / 2 ≠ ↑(1 / 2 : ℝ)
+    rw [ofReal_div, ofReal_one, ofReal_ofNat]
+    exact hs' ∘ (div_left_inj' two_ne_zero).mp
+
+lemma differentiable_completedHurwitzZetaEven₀ (a : UnitAddCircle) :
+    Differentiable ℂ (completedHurwitzZetaEven₀ a) :=
+  ((hurwitzEvenFEPair a).differentiable_Λ₀.comp (differentiable_id.div_const _)).div_const _
+
+/-- The difference of two completed even Hurwitz zeta functions is differentiable at `s = 1`. -/
+lemma differentiableAt_one_completedHurwitzZetaEven_sub_completedHurwitzZetaEven
+    (a b : UnitAddCircle) :
+    DifferentiableAt ℂ (fun s ↦ completedHurwitzZetaEven a s - completedHurwitzZetaEven b s) 1 := by
+  have (s) : completedHurwitzZetaEven a s - completedHurwitzZetaEven b s =
+      completedHurwitzZetaEven₀ a s - completedHurwitzZetaEven₀ b s -
+      ((if a = 0 then 1 else 0) - (if b = 0 then 1 else 0)) / s := by
+    simp_rw [completedHurwitzZetaEven_eq, sub_div]
+    abel
+  rw [funext this]
+  refine .sub ?_ <| (differentiable_const _ _).div (differentiable_id _) one_ne_zero
+  apply DifferentiableAt.sub <;> apply differentiable_completedHurwitzZetaEven₀
+
+lemma differentiableAt_completedCosZeta
+    (a : UnitAddCircle) {s : ℂ} (hs : s ≠ 0) (hs' : s ≠ 1 ∨ a ≠ 0) :
+    DifferentiableAt ℂ (completedCosZeta a) s := by
+  refine (((hurwitzEvenFEPair a).symm.differentiableAt_Λ (Or.inl ?_) ?_).comp s
+      (differentiableAt_id.div_const _)).div_const _
+  · rwa [div_ne_zero_iff, eq_true_intro two_ne_zero, and_true]
+  · change s / 2 ≠ ↑(1 / 2 : ℝ) ∨ (if a = 0 then 1 else 0) = 0
+    refine Or.imp (fun h ↦ ?_) (fun ha ↦ ?_) hs'
+    · simpa only [push_cast] using h ∘ (div_left_inj' two_ne_zero).mp
+    · simp_rw [eq_false_intro ha, if_false]
+
+lemma differentiable_completedCosZeta₀ (a : UnitAddCircle) :
+    Differentiable ℂ (completedCosZeta₀ a) :=
+  ((hurwitzEvenFEPair a).symm.differentiable_Λ₀.comp (differentiable_id.div_const _)).div_const _
+
+/-- The residue of `completedHurwitzZetaEven a s` at `s = 1` is equal to `1`. -/
+lemma completedHurwitzZetaEven_residue_one (a : UnitAddCircle) :
+    Tendsto (fun s ↦ (s - 1) * completedHurwitzZetaEven a s) (𝓝[≠] 1) (𝓝 1) := by
+  have h1 : Tendsto (fun s : ℂ ↦ (s - ↑(1  / 2 : ℝ)) * _) (𝓝[≠] ↑(1  / 2 : ℝ))
+    (𝓝 ((1 : ℂ) * (1 : ℂ))) := (hurwitzEvenFEPair a).Λ_residue_k
+  simp only [push_cast, one_mul] at h1
+  have h2 : Tendsto (fun s : ℂ ↦ s / 2) (𝓝[≠] 1) (𝓝[≠] (1 / 2)) :=
```

Comment:
A version of this occurs three times here. Maybe add a
```lean
lemma xyz {a : ℂ} (b : ℂ) (ha : a ≠ 0) : Tendsto (fun s : ℂ ↦ s / a) (𝓝[≠] b) (𝓝[≠] (b / a))
```
somewhere? (Or state it as a private lemma for `a = 2`.)